client/checkout/utils/upe.js

```diff
@@ -183,17 +202,24 @@ export const appendFraudPreventionTokenInputToForm = ( $form ) => {
  * @return {boolean} Boolean indicating whether a saved payment method is being used.
  */
 export function isUsingSavedPaymentMethod( paymentMethodType ) {
-	const prefix = '#wc-woocommerce_payments';
-	const suffix = '-payment-token-new';
-	const savedPaymentSelector =
-		paymentMethodType === 'card' || paymentMethodType === 'link'
-			? prefix + suffix
-			: prefix + '_' + paymentMethodType + suffix;
+	const paymentsForm = document.querySelector(
+		`.wcpay-upe-form[data-payment-method-type="${ paymentMethodType }"]`
+	);
+	if ( ! paymentsForm ) {
+		return false;
+	}
 
-	return (
-		document.querySelector( savedPaymentSelector ) !== null &&
-		! document.querySelector( savedPaymentSelector ).checked
+	const newPaymentTokenInputId = `wc-${ getGatewayIdBy(
+		paymentMethodType
+	) }-payment-token-new`;
+	const newPaymentTokenInput = paymentsForm.querySelector(
+		`input#${ newPaymentTokenInputId }`
 	);
+	if ( ! newPaymentTokenInput ) {
+		return false;
+	}
+
+	return newPaymentTokenInput !== null && ! newPaymentTokenInput.checked;
```

Comment:
```suggestion
	return ! newPaymentTokenInput.checked;
```

Since `! newPaymentTokenInput` is already being checked above, we shouldn't need to `!== null`